### PR TITLE
feat: add WebView::bounds getter

### PR DESCRIPTION
**Pull request by lucasfernog**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### Checklist
- [ ] This PR will resolve #___
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary
- [ ] It can be built on all targets and pass CI/CD.

### CODE REVIEW

File: src/webview2/mod.rs [hunk 1]
```diff
@@ -1009,6 +1010,35 @@ impl InnerWebView {
...
+        width: (rect.right - rect.left) as u32,
+        height: (rect.bottom - rect.top) as u32,
+      }
+    } else {
+      Rect {
+        x: 0,
+        y: 0,
+        width: 0,
+        height: 0,
+      }
```
Comment (amrbashir) [inline src/webview2/mod.rs]: we can return `self.controller.bounds` here.

File: src/wkwebview/mod.rs [hunk 2]
```diff
@@ -1090,6 +1090,30 @@ r#"Object.defineProperty(window, 'ipc', {
     Ok(())
   }
 
+  pub fn bounds(&self) -> Rect {
+    if self.is_child {
```
Comment (amrbashir) [inline src/wkwebview/mod.rs]: I believe we can run the same logic whether it is child or not or with a small tweak.

File: src/webkitgtk/mod.rs [hunk 3]
```diff
@@ -600,6 +600,68 @@ impl InnerWebView {
...
+    let mut bounds = Rect {
+      x: 0,
+      y: 0,
+      width: 0,
+      height: 0,
+    };
+
+    if let (Some(xlib), Some(display), Some(window_handle)) =
+      (&self.xlib, self.x11_display, self.x11_window)
+    {
```
Comment (amrbashir) [inline src/webkitgtk/mod.rs:608]: We can probably skip using x11 and use `self.gtk_window` directly if it is a child

File: src/webkitgtk/mod.rs [hunk 4]
```diff
@@ -600,6 +600,68 @@ impl InnerWebView {
...
+        }
+      }
+    } else if let Some(window) = &self.gtk_window {
+      if self.is_child {
+        let position = window.position();
+        bounds.x = position.0;
+        bounds.y = position.1;
+      }
+
+      let size = window.size();
```
Comment (amrbashir) [inline src/webkitgtk/mod.rs:625]: if we are not a child, we should be using `self.webview.allocated_size` instead

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: src/webkitgtk/mod.rs (hunk 3 above)
Comment (lucasfernog) [reply src/webkitgtk/mod.rs:608]: I tried that, but I use i3wm and by default it expands the window to fill the tile, so the bounds wrong without using x11. if we don't really care about tiling window managers then yeah we can skip to gtk directly

**Comment by lucasfernog** [comment]
also fixed an issue with the webview size on macOS. let me know if you want a separate PR

File: src/webkitgtk/mod.rs (hunk 3 above)
Comment (amrbashir) [reply src/webkitgtk/mod.rs:608]: it is fine to keep then

**Review by amrbashir — APPROVED** [review]

**Pull request merged**